### PR TITLE
SBERDOMA-952 Division list page

### PR DESCRIPTION
**Pull request by MrFoxPro**
```
FEATURE_FLAGS_CONFIG='["divisions"]'
```
image
image

### CODE REVIEW

File: apps/condo/domains/division/components/DivisionForm/CreateDivisionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            {({ handleSave, isLoading }) => {
+                return (
+                    <Form.Item noStyle dependencies={['properties', 'responsible']}>
+                        {
+                            ({ getFieldsValue }) => {
+                                const { properties, responsible } = getFieldsValue(['properties', 'responsible'])
+
+                                return (
+                                    <ActionBar>
+                                        <Space size={12}>
```
Comment (SavelevMatthew) [inline apps/condo/domains/division/components/DivisionForm/CreateDivisionForm.tsx:49]: antd Space is not treat well with adaptivity out of the box...so...make sure to check this before merging

File: apps/condo/domains/division/hooks/useTableColumns.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    setFiltersApplied: React.Dispatch<React.SetStateAction<boolean>>): Array<ITableColumn> => {
+    const intl = useIntl()
+    const columns = useMemo(() => {
+        const DivisionTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Division' })
+        const BuildingsTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Buildings' })
+        const ForemanTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Foreman' })
+        const TechiesTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Techies' })
+        // const sorterMap = createSorterMap(sort)
+        const search = getFilteredValue(filters, 'search')
+        const render = (text, isArray = false) => {
```
Comment (SavelevMatthew) [inline apps/condo/domains/division/hooks/useTableColumns.tsx:41]: Use common render as soon as https://github.com/open-condo-software/condo/pull/498 will go to master
Will be placed in `@condo/domains/common/components/Table/Renders`

File: apps/condo/domains/division/utils/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export const searchToQuery = (search?: string): DivisionWhereInput[] => {
+    if (!search) {
+        return
+    }
+
+    return [
+        { properties_some: { address_contains_i:  search } },
+    ]
+}
+export const filtersToQuery = (filters: IFilters): DivisionWhereInput => {
```
Comment (SavelevMatthew) [inline apps/condo/domains/division/utils/helpers.ts:15]: Please refer to this manual (https://doma.atlassian.net/wiki/spaces/SBER/pages/787808257/1) after this (https://github.com/open-condo-software/condo/pull/498) PR will come to master...
I guess you will need to create a custom filter for properties _some, but for address `getStringContainsFilter('address')` will be enough

File: apps/condo/pages/property/index.tsx [hunk 4]
```diff
@@ -60,47 +77,96 @@ export const PropertyPageViewMap = ({ searchPropertyQuery }): React.FC => {
...
+        filtersState={filtersState}
+        createRoute="/division/create"
+        objects={objects}
+        generateExcelData={generateExcelData}
+        handleRowAction={handleRowAction}
+    />
+}
+
+
+const BuildingsPageViewTable = () => {
```
Comment (SavelevMatthew) [inline apps/condo/pages/property/index.tsx:147]: U guess it should be moved to components of property domain

**Review by SavelevMatthew — CHANGES_REQUESTED** [review]

File: apps/condo/domains/division/components/DivisionsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+                router.push(`/division/${record.id}/`)
+            },
+        }
+    }
+
+    // TODO(mrfoxpro): move to common
+    const applyFiltersToQuery = (newFilters) => {
+        const query = { ...router.query, filters: JSON.stringify(newFilters) }
+        const newQuery = qs.stringify({ ...query }, { arrayFormat: 'comma', skipNulls: true, addQueryPrefix: true })
+        console.log(router.route)
```
Comment (AntonAL) [inline apps/condo/domains/division/components/DivisionsTable.tsx:113]:
```suggestion
        
```

File: apps/condo/domains/property/utils/clientSchema/Property.ts [hunk 6]
```diff
@@ -17,6 +17,9 @@ import { BuildingMap } from '@condo/domains/property/components/panels/Builder/M
 export interface IPropertyUIState extends Property {
     id: string
     address: string
+    addressMeta: {
+        data: any
```
Comment (AntonAL) [inline apps/condo/domains/property/utils/clientSchema/Property.ts:21]:
```suggestion
        data: AddressMeta
```

File: apps/condo/lang/en.json [hunk 7]
```diff
@@ -107,7 +107,7 @@
   "menu.ControlRoom": "Control Room",
   "menu.Employee": "Employee",
   "menu.Home": "Dashboard",
-  "menu.Property": "Buildings/objects",
+  "menu.Property": "Buildings and Divisions",
```
Comment (AntonAL) [inline apps/condo/lang/en.json:110]:
```suggestion
  "menu.Property": "Buildings & Divisions",
```

File: apps/condo/lang/en.json [hunk 8]
```diff
@@ -257,9 +257,15 @@
   "pages.condo.property.index.TableField.Address": "Address",
   "pages.condo.property.index.TableField.TasksInWorkCount": "Tickets in work",
   "pages.condo.property.index.TableField.UnitsCount": "Flats",
+  "pages.condo.property.index.TableField.Division": "Name of division",
+  "pages.condo.property.index.TableField.Buildings": "Buildings",
+  "pages.condo.property.index.TableField.Foreman": "Foreman",
+  "pages.condo.property.index.TableField.Techies": "Techies",
```
Comment (AntonAL) [inline apps/condo/lang/en.json:263]:
```suggestion
  "pages.condo.property.index.TableField.Technicians": "Technicians",
```

File: apps/condo/lang/ru.json [hunk 9]
```diff
@@ -257,9 +257,15 @@
   "pages.condo.property.index.TableField.Address": "Адрес",
   "pages.condo.property.index.TableField.TasksInWorkCount": "Заявок в работе",
   "pages.condo.property.index.TableField.UnitsCount": "Квартир",
+  "pages.condo.property.index.TableField.Division": "Название участка",
+  "pages.condo.property.index.TableField.Buildings": "Дома",
+  "pages.condo.property.index.TableField.Foreman": "Мастер",
+  "pages.condo.property.index.TableField.Techies": "Техники",
```
Comment (AntonAL) [inline apps/condo/lang/ru.json:263]:
```suggestion
  "pages.condo.property.index.TableField.Technicians": "Техники",
```

File: apps/condo/domains/division/hooks/useTableColumns.tsx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+    filterDropdown?: unknown,
+    filterIcon?: unknown
+}
+
+export const useTableColumns = (sorters: Sorters, filters: FiltersFromQueryType) => {
+    const intl = useIntl()
+    const DivisionTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Division' })
+    const BuildingsTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Buildings' })
+    const ForemanTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Foreman' })
+    const TechiesTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Techies' })
```
Comment (AntonAL) [inline apps/condo/domains/division/hooks/useTableColumns.tsx:33]:
```suggestion
    const TechniciansTitleMessage = intl.formatMessage({ id: 'pages.condo.property.index.TableField.Technicians' })
```

File: apps/condo/domains/division/hooks/useTableColumns.tsx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+                title: ForemanTitleMessage,
+                ellipsis: true,
+                dataIndex: 'responsible',
+                key: 'responsible',
+                render: (responsible) => render(responsible.name),
+
+                width: 20,
+            },
+            {
+                title: TechiesTitleMessage,
```
Comment (AntonAL) [inline apps/condo/domains/division/hooks/useTableColumns.tsx:89]:
```suggestion
                title: TechniciansTitleMessage,
```

File: apps/condo/domains/property/utils/clientSchema/Property.ts (hunk 6 above)
Comment (SavelevMatthew) [reply apps/condo/domains/property/utils/clientSchema/Property.ts:21]: Data is not of type of AdressMeta, it's just field of it....So I guess it should be `addressMeta: AddressMeta`
You can find it in `@condo/domains/common/utils/addressApi/AddressMeta`

**Comment by MrFoxPro** [comment]
Changes moved to https://github.com/open-condo-software/condo/pull/500

**Pull request closed without merging**